modules/xfem/include/materials/LevelSetBiMaterial.h

```diff
@@ -0,0 +1,50 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef LEVELSETBIMATERIAL_H
+#define LEVELSETBIMATERIAL_H
+
+#include "Material.h"
+
+// Forward Declarations
+class LevelSetBiMaterial;
+class XFEM;
+
+template <>
+InputParameters validParams<LevelSetBiMaterial>();
+
+/**
+ * Compute a material property for bi-materials defined by a level set function.
+ */
+class LevelSetBiMaterial : public Material
+{
+public:
+  LevelSetBiMaterial(const InputParameters & parameters);
+
+protected:
+  virtual void computeProperties();
+  virtual void computeQpProperties();
+
+  virtual void computeQpPropertiesForLevelSetNegative() = 0;
+  virtual void computeQpPropertiesForLevelSetPositive() = 0;
+
+  // global material properties
+  std::string _base_name;
+
+  std::shared_ptr<XFEM> _xfem;
+
+  /// name of level set variable
+  NonlinearVariableName _ls_var_name;
+
+  /// level set variable
+  MooseVariable & _ls_var;
```

Comment:
Please use `_level_set_var_name` and `_level_set_var` for the variable names for clarity in the code